backend/app/routers/routers.py

```diff
@@ -0,0 +1,42 @@
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Request
+
+from app.models.record import Record
+from app.routers.dependencies import next_records_parameters
+
+router = APIRouter(prefix="/ap1/v1/nerdswipe", tags=["/api/v1/nerdswipe"])
+
+
+@router.get("/next_records")
+async def next_records(
+    next_params: Annotated[dict, Depends(next_records_parameters)], request: Request
+) -> tuple[Record, ...]:
+    """
+    This is the path operation function of the /next endpoint.
+    Returns a tuple of Records from the AntiRecommendationEngine.
+    """
+
+    return request.app.state.anti_recommendation_engine.get_next_records(
+        next_params["record_key"], next_params["record_type"]
```

Comment:
So this is why you have `str` on those methods. You should impose stronger types / do validation at the boundary here.